activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -303,8 +303,14 @@ def execute_grouped_calculation(operation, column_name, distinct) #:nodoc:
       calculated_data = @klass.connection.select_all(relation, nil, relation.bound_attributes)
 
       if association
-        key_ids     = calculated_data.collect { |row| row[group_aliases.first] }
-        key_records = association.klass.base_class.where(association.klass.base_class.primary_key => key_ids)
+        relation_class = association.klass.base_class
+        key_ids = calculated_data.collect { |row| row[group_aliases.first] }
+        nils, key_ids = key_ids.partition(&:nil?)
```

Comment:
We can leave the `nil`s in `key_ids` - clarifies that we'll pass to `IN` verbatim, but do some additional deprecated handling for `OR … IS NULL`.